src/Functions/if.cpp

```diff
@@ -888,6 +940,27 @@ class FunctionIf : public FunctionIfBase
         return nullptr;
     }
 
+    static void executeShortCircuitArguments(ColumnsWithTypeAndName & arguments)
+    {
+        int last_short_circuit_argument_index = checkShirtCircuitArguments(arguments);
+        if (last_short_circuit_argument_index == -1)
+            return;
+
+        /// Check if condition is const or null to not create full mask from it.
+        if ((isColumnConst(*arguments[0].column) || arguments[0].column->onlyNull()) && !arguments[0].column->empty())
+        {
+            bool value = arguments[0].column->getBool(0);
+            executeColumnIfNeeded(arguments[1], !value);
+            executeColumnIfNeeded(arguments[2], value);
+            return;
+        }
+
+        IColumn::Filter mask(arguments[0].column->size(), 1);
+        auto mask_info = extractMaskInplace(mask, arguments[0].column);
+        maskedExecute(arguments[1], mask, mask_info);
+        maskedExecute(arguments[2], mask, mask_info, /*inverted=*/true);
```

Comment:
I think, it is just easier to invert mask here, and remove `inveted` flag from `maskedExecute`. (as well as remove inverted flag from `ICoulmn::filter`)